acceptance-tests/src/main/java/hudson/plugin/gradle/ath/config/WebDriverProvider.java

```diff
@@ -0,0 +1,97 @@
+package hudson.plugin.gradle.ath.config;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import org.jenkinsci.test.acceptance.FallbackConfig;
+import org.jenkinsci.test.acceptance.guice.TestCleaner;
+import org.jenkinsci.test.acceptance.guice.TestName;
+import org.jenkinsci.test.acceptance.selenium.Scroller;
+import org.jenkinsci.test.acceptance.utils.ElasticTime;
+import org.junit.runners.model.Statement;
+import org.openqa.selenium.Dimension;
+import org.openqa.selenium.UnsupportedCommandException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.support.events.EventFiringWebDriver;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Locale;
+import java.util.logging.Logger;
+
+class WebDriverProvider implements Provider<WebDriver> {
+
+    private static final Logger LOGGER = Logger.getLogger(FallbackConfig.class.getName());
+    private final TestCleaner testCleaner;
+    private final FallbackConfig fallbackConfig;
+    private final TestName testName;
+    private final ElasticTime time;
+
+    @Inject
+    public WebDriverProvider(
+        TestCleaner testCleaner,
+        FallbackConfig fallbackConfig,
+        TestName testName,
+        ElasticTime time) {
+        this.testCleaner = testCleaner;
+        this.fallbackConfig = fallbackConfig;
+        this.testName = testName;
+        this.time = time;
+    }
+
+    private String getBrowser() {
+        String browser = System.getenv("BROWSER");
+        if (browser == null) browser = "chrome-container";
+        browser = browser.toLowerCase(Locale.ENGLISH);
+        return browser;
+    }
+
+    @Override
+    public WebDriver get() {
+        if ("chrome".equals(getBrowser())) {
+            return createChromeWebDriver();
+        }
+        try {
+            return fallbackConfig.createWebDriver(testCleaner, testName, time);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private WebDriver createChromeWebDriver() {
+        ChromeOptions chromeOptions = new ChromeOptions();
+        chromeOptions.setBinary("/usr/bin/google-chrome-for-testing");
```

Comment:
Using `/usr/bin/google-chrome-for-testing` would point you to the latest version and as soon as we would update this would conflict with your chromedriver version.

I would change this to: `/opt/google/119.0.6045.105/chrome-linux64/chrome`

```
zielezin@dev34:~$ /opt/google/119.0.6045.105/chrome-linux64/chrome --version
Google Chrome for Testing 119.0.6045.105
```

Reply:
So actually, after investigating, it turned out we can delegate all the driver plumbing to Selenium. It will try to find the specified Chrome version installed and if not found, download it. So my latest commit specifies the version to `stable` which should afaik try to find a stable Chrome installed, and if not present will download the latest stable.